torchx/lib/torchx/backend.ex

```diff
@@ -0,0 +1,219 @@
+defmodule Torchx.Backend do
+  @behaviour Nx.Backend
+
+  defstruct [:ref]
+
+  alias Torchx.NIF
+
+  def torch_type({:u, 8}), do: :byte
+
+  def torch_type({:u, 16}),
+    do: raise(ArgumentError, "PyTorch does not support unsigned 16 bit integer.")
+
+  def torch_type({:u, 32}),
+    do: raise(ArgumentError, "PyTorch does not support unsigned 32 bit integer.")
+
+  def torch_type({:u, 64}),
+    do: raise(ArgumentError, "PyTorch does not support unsigned 64 bit integer.")
+
+  def torch_type({:s, 8}), do: :char
+  def torch_type({:s, 16}), do: :short
+  def torch_type({:s, 32}), do: :int
+  def torch_type({:s, 64}), do: :long
+  def torch_type({:bf, 16}), do: :brain
+  def torch_type({:f, 16}), do: :half
+  def torch_type({:f, 32}), do: :float
+  def torch_type({:f, 64}), do: :double
+
+  ## Creation
+
+  @impl true
+  def eye(%{shape: {n, n}, type: type} = out) do
```

Comment:
Let's pattern match on the struct in all of the function heads below. It will help us find bugs (like mistyped field names):

```suggestion
  alias Nx.Tensor, as: T

  def eye(%T{shape: {n, n}, type: type} = out) do
```